notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/Installation.java

```diff
@@ -88,30 +92,44 @@ public void clearTags() {
     }
 
     /**
-     *
-     * @param tag
-     * @return
+     * Remove specific tag from this Installation.
+     * @param tag The tag that should no longer be in the collection.
+     * @return True if the tag had previously been associated with this collection.
      */
     public boolean removeTag(String tag) {
         return mTags.remove(tag);
     }
 
+    /**
+     * Deletes several tags from this collection.
+     * @param tags The tags that should no longer be in the collection.
+     * @return True if any of the tags had previously been associated with this collection.
+     */
     public boolean removeTags(Collection<? extends String> tags) {
         return mTags.removeAll(tags);
     }
 
+    /**
+     * Adds several tamplates to this Installation.
+     * @param templates The templates to include with this Installation.
+     */
+    public void addTemplates(Map<String, InstallationTemplate> templates) { mTemplates.putAll(templates); }

```

Comment:
I'm not sold between add templates or set templates, meaning can you overwrite a template that already exists?

Reply:
This is a good point. Only having `addTemplates` merges what's there, but is somewhat ambiguous around behavior when there is an existing template for a given name.

Reply:
Chatted offline - we're going to review this and other differences that have cropped up early next week. We'll fix this and any other issues that pop up for preview3.